src/llama.cpp

```diff
@@ -14154,9 +14164,19 @@ static void llama_set_inputs(llama_context & lctx, const llama_batch & batch) {
                     }
                 }
 
-                for (int i = n_tokens; i < GGML_PAD(n_tokens, GGML_KQ_MASK_PAD); ++i) {
-                    for (int j = 0; j < n_kv; ++j) {
-                        data[h*(n_kv*n_tokens) + i*n_kv + j] = -INFINITY;
+                if (data) {
+                    for (int i = n_tokens; i < GGML_PAD(n_tokens, GGML_KQ_MASK_PAD); ++i) {
+                        for (int j = 0; j < n_kv; ++j) {
+                            data[h*(n_kv*n_tokens) + i*n_kv + j] = -INFINITY;
+                        }
+                    }
+                }
+
+                if (data_swa) {
```

Comment:
`data_swa` is also modified by the code above (line 14158). It is used by gemma2.

Overwriting it here may break gemma2

Reply:
Sorry I'm not familiar with gemma2, so I haven't test the PR on gemma2. I only test this PR with Phi3 on CPU.

I do not understand when it should padded to `GGML_KD_MASK_PAD` or not. I pad `data_swa` to GGML_KD_MASK_PAD because it looks like the original just forgets to do so. Actually, padding `data_swa` or not does not affect the correctness of Phi3 on CPU.

I'm confused on the original code that `data` is explicitly padded to `GGML_KQ_MASK_PAD` but `data_swa` is not. Is this the intended behavior? If yes, I'm happy to revert the change (padding data_swa to GGML_KQ_MASK_PAD). but I still want someone could explain to me what `GGML_KQ_MASK_PAD` actually means.

Reply:
@ngxson I agree with @FanShupei here, I think `data_swa` should also be padded. I don't see why not, since the ranges of data written here and above do not overlap.

Not sure why this worked before though. Padding `data_swa` seems saner than leaving the values uninitialized.

Reply:
Both should be padded. The padding is necessary so that GPU kernels (such as the Metal Flash-Attention) not perform extra checks for out-of-bounds access when working on chunks of data